### ISSUE TITLE
Can't detect lying people after Training

### ISSUE THREAD

**Issue by TobiasVorsmann** [AlexeyAB/darknet#7267]
Hello community.

I am trying to train a yolo-tiny-v4 to detect persons. I am using a fisheye camera to get an image which I want to process. I have a big dataset with more than 200k images in the format of 1280 x 720 (all of them are distorted to match my camera).  The problem is, that after the network was trained I am not able to detect people lying on the ground, just standing or sitting people are possible. I have already checked the labels and they fit the images. Furthermore there are a lot of transverse anchors so the people should fit in there (see attachment below)
![anchors](https://user-images.githubusercontent.com/43747781/105379917-9dff0c00-5c0d-11eb-90e6-2deec72aa0e7.png)

The train and test dataset both contain a lot of standing and lying people so there should be no problem in detection.
I have also tried to train the network with only one class (person) or multiple classes like (person, pottedplant, car, ...) but same results for all of them, no lying persons (standing persons are working very well).

Another point I would like to mention is that the loss is decreasing really fast (too fast in my opinion). If I decrease the learning rate, the mAP calculation takes forever (literally, it is like 2 or 3 hours for one calculation). I found a workaround (using the steps and chosing steps 1,2,3 for example to decrease the learning rate) but that is not the right way i guess.

![chart](https://user-images.githubusercontent.com/43747781/105382280-2e3e5080-5c10-11eb-9f25-c2cf2914087d.png)

I am training with a network resolution of 608 x 608 because the mAP calculation is not working if the network size is set to 608 x 352. Below is my configuration file:
[yolov4-tiny.txt.txt](https://github.com/AlexeyAB/darknet/files/5850797/yolov4-tiny.txt.txt)

I don't know what I am doing wrong and would be very thankful if anybody could help me solving this problem.
@AlexeyAB Do you have any tips or advices for me?

Thanks in advance to anybody! :)

**Comment by blackwool** [comment]
Would you please show some sample pictures (neg and pos) and labels file as examples  ？

**Comment by TobiasVorsmann** [comment]
Hi @blackwool!

Thanks for your reply. Attached are sample images and the labelfiles. Negative images are completely different objects with empty lable files.

Could you explain your thoughts about that? Why do you need the files? I have already checked, that they are fitting or am I missing something?
Thanks for the help! :)

![sample1](https://user-images.githubusercontent.com/43747781/106491997-bd245600-64b7-11eb-8680-6cc3dd4ce4ca.jpg)
[sample1.txt](https://github.com/AlexeyAB/darknet/files/5905427/sample1.txt)
![sample2](https://user-images.githubusercontent.com/43747781/106492000-beee1980-64b7-11eb-81f9-d8d3add403dc.jpg)
[sample2.txt](https://github.com/AlexeyAB/darknet/files/5905430/sample2.txt)

**Comment by versavel** [comment]
Your label files are not correct. Each line should have 5 items: id, x, y, w, h
Where x and y are coordinates of center bounding box, w and h the width and height of the bounding box.

**Comment by blackwool** [comment]
Pieter versavel answer is right. You should check each label file .

sample2.txt file is not correct. 
sample1.txt file is right .

**Comment by versavel** [comment]
Tobias, I looked at your annotation files again, this time on my computer, and they actually look fine. Earlier I was looking at them on my phone and due to line wrapping a single line looked like two (incorrect) lines.

**Comment by versavel** [comment]
What was your motivation to use 20, rather than the usual 6, anchor boxes in your yolo-v4-tiny network ?

**Comment by blackwool** [comment]
You use 20 anchors ,then the num should also be changed to  20  in each yolo layer config.

**Comment by blackwool** [comment]
You use 20 anchors ,then the num should also be changed to 20 in each yolo layer config.

**Comment by TobiasVorsmann** [comment]
@versavel  Ok, at least the files are fine :) I am using 20 anchors to get a higher accuraccy. There are a lot of different shapes for the boxes. I used the detector calc_anchors and tried out different numbers of anchors. With 6 anchors I got a too low IoU. 20 anchors seems to be the best amount of anchors for me.

@blackwool For my understanding I only have to change the num parameter if I don't set the mask ("If mask is absence, then filters=(classes + coords + 1)*num)" see [How to train (to detect your custom objects)](https://github.com/AlexeyAB/darknet#how-to-train-to-detect-your-custom-objects) section). For other trainings I didn't changed the num parameter and they worked quite well. Do I have to change the num parameter for sure or only if I don't use the mask?

**Comment by blackwool** [comment]
Change the num paramter first.
because in yolo_layer.c

 line 33:  l.biases = (float*)xcalloc(_total_ * 2, sizeof(float)); 

 _total_ value is equal to  num.
 _l.biases_ array will be used for each mask ( from 0 to 19)

**Comment by blackwool** [comment]
You can see yolo4.cfg. the  num =9  ,equals to anchors number

**Comment by TobiasVorsmann** [comment]
@blackwool Thanks for your reply. I changed the num parameter to 20 and started another training. Will see if this solves my problem ;)

**Comment by TobiasVorsmann** [comment]
Thanks for the help, works better than before. I can detect people as long as they are not lying flat on the ground, which is quiet confusing because there are images in the dataset covering this scenario. Any idea how to become better?

**Comment by versavel** [comment]
How many images in the training dataset have the "lying flat on the ground" object ? 
If it's a small number compared to the other objects, and you have a large enough dataset, you could remove images from the training set to increase the relative number of "lying flat on the ground" objects in your training dataset. 
Alternatively, if your training dataset is on the smaller side, you could add "augmented images" that have a  "lying flat on the ground" object to the training dataset. For instance, for each image which has the "lying flat on the ground" object in it, you could create 5 additional images where the object was randomly translated up/down/left/right.

**Comment by blackwool** [comment]
When training,the image will be fliped or rotated for data argumention. I think that lying people and standing people will will be confused.  ------------------&nbsp;Original&nbsp;------------------

**Comment by stephanecharette** [comment]
Darknet will only flip left->right.  It will not rotate images.  Check to see if `flip=0` or `flip=1` has been set in your .cfg file.  Set to zero to disable, or 1 to enable.  I don't recall what the default value is for "flip".

**Comment by Micky-123** [comment]
@TobiasVorsmann : I have a similar use case where eventually I want person detection in standing, sitting and lying postures (in phases - first sitting and standing and then lying)  At this point, I am not done fully with my dataset. But would like to check with you as whether with tiny model - you were eventually able to detect in different postures? And did you make any changes in the tiny cfg file? Were you able to detect person very close to camera to a distance of 4-5 meters? Any inputs will be helpful for me.
